vms/avm/block/builder/builder.go

```diff
@@ -93,12 +93,12 @@ func (b *builder) BuildBlock(context.Context) (snowman.Block, error) {
 		remainingSize = targetBlockSize
 	)
 	for {
-		tx := b.mempool.Peek()
+		tx, exists := b.mempool.Peek()
 		// Invariant: [mempool.MaxTxSize] < [targetBlockSize]. This guarantees
 		// that we will only stop building a block once there are no
 		// transactions in the mempool or the block is at least
 		// [targetBlockSize - mempool.MaxTxSize] bytes full.
-		if tx == nil || len(tx.Bytes()) > remainingSize {
+		if !exists && tx == nil || len(tx.Bytes()) > remainingSize {
```

Comment:
I don't think we should check if tx == nil anymore

Reply:
I thought it could still return nil but we do guard against that since we verify the tx first before adding it to the mempool- dropping